### PR TITLE
 agent creates ticket notification

### PR DESCRIPTION
**Pull request by hukaixuan2017**
Add agent who creates ticket notification.

Xian Dian Tong Software Co., Ltd., http://www.dian-tong.com/

### CODE REVIEW

File: Kernel/Modules/AdminNotificationEvent.pm [hunk 1]
```diff
@@ -784,7 +785,6 @@ sub _Edit {
             AgentMyQueuesMyServices   => Translatable('All agents subscribed to both the ticket\'s queue and service'),
             Customer                  => Translatable('Customer user of the ticket'),
             AllRecipientsFirstArticle => Translatable('All recipients of the first article'),
-            AllRecipientsLastArticle  => Translatable('All recipients of the last article'),
```
Comment (mgruner) [inline Kernel/Modules/AdminNotificationEvent.pm]: This removal was probably unintended, right?

File: Kernel/Modules/AdminNotificationEvent.pm [hunk 2]
```diff
@@ -775,6 +775,7 @@ sub _Edit {
 
     $Param{RecipientsStrg} = $LayoutObject->BuildSelection(
         Data => {
+            AgentCreate               => Translatable('Agent who creates the ticket'),
```
Comment (mgruner) [inline Kernel/Modules/AdminNotificationEvent.pm]: Should we rather say "who create*d* the ticket"?

**Comment by hukaixuan2017** [comment]
@mgruner  I am glad to receive your feed back.
> @@ -784,7 +785,6 @@ sub _Edit {
             AgentMyQueuesMyServices   => Translatable('All agents subscribed to both the ticket\'s queue and service'),
             Customer                  => Translatable('Customer user of the ticket'),
             AllRecipientsFirstArticle => Translatable('All recipients of the first article'),
-            AllRecipientsLastArticle  => Translatable('All recipients of the last article'), this line was accidentally deleted by me, it my falut. I will be add this line and unit test to pull again.     $Param{RecipientsStrg} = $LayoutObject->BuildSelection(
         Data => {
+            AgentCreate               => Translatable('Agent who creates the ticket'),

If customer create ticket , the notificatoin will not be sent to customer. so we sya "Agent who creates the ticket".



hu.kaixuan@dian-tong.com
 
From: Martin Gruner
Date: 2017-07-05 15:17
To: OTRS/otrs
CC: hukaixuan2017; Author
Subject: Re: [OTRS/otrs] agent creates ticket notification (#1786)
@mgruner requested changes on this pull request.
This looks generally nice. Could you please add test cases to scripts/test/Ticket/Event/NotificationEvent.t to verify proper handling of the new recipient selection?


In Kernel/Modules/AdminNotificationEvent.pm:
This removal was probably unintended, right?


In Kernel/Modules/AdminNotificationEvent.pm:
> @@ -775,6 +775,7 @@ sub _Edit {
 
     $Param{RecipientsStrg} = $LayoutObject->BuildSelection(
         Data => {
+            AgentCreate               => Translatable('Agent who creates the ticket'),

Should we rather say "who created the ticket"?

**Comment by mgruner** [comment]
That's a good point. What happens if a customer creates the ticket? Then OTRS uses `CustomerPanelUserID` internally, which typically is set to 1 (root user). Will in this case `AgentCreate` also refer to this user and send a notification there? It should probably only happen if the SenderType of the first article is also `Agent`, right?

**Comment by hukaixuan2017** [comment]
You mean we need to set a filter. If the first article of the ticket has the SenderType of Agent, then the notification should be sent to the creator ?



hu.kaixuan@dian-tong.com
 
From: Martin Gruner
Date: 2017-07-05 17:24
To: OTRS/otrs
CC: hukaixuan2017; Author
Subject: Re: [OTRS/otrs] agent creates ticket notification (#1786)
That's a good point. What happens if a customer creates the ticket? Then OTRS uses CustomerPanelUserID internally, which typically is set to 1 (root user). Will in this case AgentCreate also refer to this user and send a notification there? It should probably only happen if the SenderType of the first article is also Agent, right?

**Comment by mgruner** [comment]
I would find this logical, what do you think? In other cases, the ticket is not created by an agent.

Martin Gruner
Team Lead R&D

OTRS AG
Bahnhofplatz 1a
94315 Straubing

T: +49 (0)6172 681988 0
F: +49 (0)9421 56818 18
I:  www.otrs.com/

Geschäftssitz: Bad Homburg, Amtsgericht: Bad Homburg, HRB 10751, USt-Nr.: DE256610065
Aufsichtsratsvorsitzender: Burchard Steinbild, Vorstand: André Mindermann (Vorsitzender), Christopher Kuhn, Sabine Riedel

____________________________________________________

OTRS 5s – Organisation.  Sicherheit.  Kommunikation.  Struktur. – Flexible Lösungen für Erfolg im Unternehmen
Link: https://www.otrs.com/?lang=de&utm_id=1032


Mehr Infos:
+ Lösungen: https://www.otrs.com/produkte/?lang=de&utm_id=1033
+ Success Stories: https://www.otrs.com/kunden/success-stories/?lang=de&utm_id=1034
+ Trainings: https://www.otrs.com/public-otrs-trainings/?lang=de&utm_id=1035

**Comment by hukaixuan2017** [comment]
I  think so too, I will do and pull it.



hu.kaixuan@dian-tong.com
 
From: Martin Gruner
Date: 2017-07-05 20:50
To: OTRS/otrs
CC: hukaixuan2017; Author
Subject: Re: [OTRS/otrs] agent creates ticket notification (#1786)
I would find this logical, what do you think? In other cases, the ticket is not created by an agent.

Martin Gruner
Team Lead R&D

OTRS AG
Bahnhofplatz 1a
94315 Straubing

T: +49 (0)6172 681988 0
F: +49 (0)9421 56818 18
I: www.otrs.com/

Geschäftssitz: Bad Homburg, Amtsgericht: Bad Homburg, HRB 10751, USt-Nr.: DE256610065
Aufsichtsratsvorsitzender: Burchard Steinbild, Vorstand: André Mindermann (Vorsitzender), Christopher Kuhn, Sabine Riedel

____________________________________________________

OTRS 5s – Organisation. Sicherheit. Kommunikation. Struktur. – Flexible Lösungen für Erfolg im Unternehmen
Link: https://www.otrs.com/?lang=de&utm_id=1032


Mehr Infos:
+ Lösungen: https://www.otrs.com/produkte/?lang=de&utm_id=1033
+ Success Stories: https://www.otrs.com/kunden/success-stories/?lang=de&utm_id=1034
+ Trainings: https://www.otrs.com/public-otrs-trainings/?lang=de&utm_id=1035

**Comment by mgruner** [comment]
Merged with changes in 2dc7d517a7de5119ea08ec5a5ef91ad0f35d1dc1, please see the diff. I added a filter to only send for tickets that were actually created by an agent (first article's sender type must be 'agent'). Thanks @hukaixuan2017!

**Pull request closed without merging**